File: docs/js/getdata.js
// Function to get URL parameters
function getUrlParams() {
    var params = {};
    var queryString = window.location.search;
    var urlParams = new URLSearchParams(queryString);
    urlParams.forEach(function(value, key) {
      params[key] = value;
    });
    return params;
}

const sheetNames = ['work', 'learn', 'team'];
const spreadsheetId = '1bQDVVO-R3tt99eR7ageBYu5XK8lnnlxHZLzJghYSLa0';
const apiKey = 'AIzaSyAUi4KazffmDZV_dQUnMUKA1jJt4i0mqlU';
let data = {};

// Function to fetch data from a specific sheet
async function fetchData(sheetName) {
  const url = `https://sheets.googleapis.com/v4/spreadsheets/${spreadsheetId}/values/${sheetName}?key=${apiKey}`;

  try {
    const response = await fetch(url);
    if (!response.ok) {
      throw new Error('Failed to fetch data');
    }
    const data = await response.json();
    // get rid of first row 
    data.values.shift();
    return data;
  } catch (error) {
    console.error('Error fetching data:', error);
    return null;
  }
}

// Function to check if data is fetched from all sheets
async function checkAndInit() {
  const promises = sheetNames.map(sheetName => fetchData(sheetName));
  const results = await Promise.all(promises);

  // Check if data is fetched from all sheets
  const allDataFetched = results.every(result => result !== null);

  if (allDataFetched) {
    // console.log('Data fetched from all sheets:', results);
    // put the data in an object
    results.forEach((result, index) => {
      data[sheetNames[index]] = result;
    });
    console.log('Data object:', data);

    init();
  } else {
    console.log('Failed to fetch data from one or more sheets.');
  }
}


// Call the function to fetch data from all sheets and initialize if successful
checkAndInit();











// // get current url to find out what page we are on
// const fullUrl = window.location.href;
// console.log(fullUrl);


// // fetch data from three different google sheets with the same id but different sheet names (work, learn, connect), then run init function

// const sheets = ['work', 'learn', 'connect']

// // fetch data from google sheets
// sheets.forEach(sheet => {
//     const apiUrl = 'https://sheets.googleapis.com/v4/spreadsheets/1bQDVVO-R3tt99eR7ageBYu5XK8lnnlxHZLzJghYSLa0/values/'+sheet+'?key=AIzaSyAUi4KazffmDZV_dQUnMUKA1jJt4i0mqlU';
//     fetch(apiUrl)
//     .then(response => {
//         // Check if the response status is OK (200)
//         if (!response.ok) {
//         throw new Error('Network response was not ok');    
//         }
//         // Parse the response as JSON
//         return response.json();
//     })    
//     .then(data => {
//         // Handle the JSON data here
//         window[sheet] = data.values
//         // get rid of first row
//         window[sheet].shift()
//         console.log(window[sheet])
//         init()
//     })    
//     .catch(error => {
//         // Handle errors
//         console.error('Fetch error:', error);
//     });    
// }    








// // const section = fullUrl.split('/')[3];


// console.log('section: ',section);

// const sheetname = section
// const apiUrl = 'https://sheets.googleapis.com/v4/spreadsheets/1bQDVVO-R3tt99eR7ageBYu5XK8lnnlxHZLzJghYSLa0/values/'+sheetname+'?key=AIzaSyAUi4KazffmDZV_dQUnMUKA1jJt4i0mqlU';

// let sectionData = null;

// // Use the fetch function to make a GET request
// fetch(apiUrl)
// .then(response => {
//     // Check if the response status is OK (200)
//     if (!response.ok) {
//     throw new Error('Network response was not ok');    
//     }
//     // Parse the response as JSON
//     return response.json();
// })    
// .then(data => {
//     // Handle the JSON data here
//     sectionData = data.values
//     // get rid of first row
//     sectionData.shift()
//     console.log(sectionData)
//     init()

// })    
// .catch(error => {
//     // Handle errors
//     console.error('Fetch error:', error);
// });    

File: docs/js/getdata.py
import requests
import json

# List of sheet names
sheet_names = ['work', 'learn', 'consult', 'musings']

# Spreadsheet ID and API key
spreadsheet_id = '1bQDVVO-R3tt99eR7ageBYu5XK8lnnlxHZLzJghYSLa0'
api_key = 'AIzaSyAxlHpEwRMRcj5qobzddd2oN9FNjWAh0RY'

# Output data dictionary
data = {}

# Function to fetch data from a specific sheet
def fetch_data(sheet_name):
    url = f'https://sheets.googleapis.com/v4/spreadsheets/{spreadsheet_id}/values/{sheet_name}?key={api_key}'
    try:
        response = requests.get(url)
        response.raise_for_status()  # Raise HTTPError for bad responses
        json_data = response.json()

        # Remove the first row (headers)
        if 'values' in json_data:
            json_data['values'].pop(0)
        return json_data
    except requests.exceptions.RequestException as e:
        print(f"Error fetching data for sheet {sheet_name}: {e}")
        return None

# Fetch data for each sheet and store it in the data dictionary
for sheet_name in sheet_names:
    sheet_data = fetch_data(sheet_name)
    if sheet_data and 'values' in sheet_data:
        data[sheet_name] = sheet_data['values']

# Save the data to a JSON file
output_file = 'sheets_data.json'
with open(output_file, 'w') as json_file:
    json.dump(data, json_file, indent=4)

print(f"Data saved to {output_file}")